graphstorm-processing/graphstorm_processing/graph_loaders/dist_heterogeneous_loader.py

```diff
@@ -1085,8 +1085,11 @@ def write_processed_feature(
                 }
                 node_type_feature_metadata[feat_name] = node_feature_metadata_dict
 
-                feat_val = single_feature_df.take(1)[0].asDict().get(feat_name, None)
-                nfeat_size = 1 if isinstance(feat_val, (int, float)) else len(feat_val)
+                feat_val = single_feature_df.take(1)[0].asDict().get(feat_name)
+                assert isinstance(
+                    feat_val, (list, numbers.Number)
+                ), "We expect features to either be scalars or lists of scalars."
```

Comment:
If feat_val is a list, shall we check whether every object in the list is a Number?

Reply:
Sure, I'll add that.